src/utils/FSHHelpers.js

```diff
@@ -236,12 +229,43 @@ function printGoFSHresults(pkg) {
   results.forEach((r) => console.log(r));
 }
 
+function addCoreFHIRVersionAndAutomaticDependencies(dependencies, coreFHIRVersion) {
+  const dependenciesToAdd = [];
+  const coreFHIRPackage = {
+    packageId: getCoreFHIRPackageIdentifier(coreFHIRVersion),
+    version: coreFHIRVersion
+  };
+  const hasCoreFHIR = hasDependency(dependencies, coreFHIRPackage);
+  if (!hasCoreFHIR) {
+    dependenciesToAdd.push(coreFHIRPackage);
+  }
+  const terminologyPkg = { packageId: 'hl7.terminology.r4', version: 'latest', isAutomatic: true };
+  const hasTerminology = hasDependency(dependencies, terminologyPkg);
+  if (!hasTerminology) {
+    dependenciesToAdd.push(terminologyPkg);
+  }
+  if (coreFHIRPackage.version.match(/^5\.0\.\d+$/)) {
+    const extensionPkg = { packageId: 'hl7.fhir.uv.extensions', version: 'latest', isAutomatic: true };
+    const hasExtensions = hasDependency(dependencies, extensionPkg);
+    if (!hasExtensions) {
+      dependenciesToAdd.push(extensionPkg);
+    }
+  }
+  return dependenciesToAdd;
+}
+
+function hasDependency(dependenciesList, currentDependency) {
+  return dependenciesList.some(
+    (dep) => dep.packageId === currentDependency.packageId && dep.version === currentDependency.version
```

Comment:
Ah that makes a lot of sense. I'll do that!

Reply:
[f5a7638](https://github.com/FSHSchool/FSHOnline/pull/134/commits/f5a7638e44e7a04dc0f5424e2187731b0971244c) updates so that if any version of either automatic dependency is specified, it won't be added to the list of dependencies. For other packages, both the id and the version are still checked so that if someone adds a core FHIR version to the list or two versions of the same package, those will still be all loaded.

Reply:
I like it!